### PR TITLE
✨ only upload files if not already on server

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
- Update `openapi.json`
- Add private `search_files` function to the files api
- When uploading a file to the server, it is checked if a file with that checksum and name exists on the server. If yes, then it is returned instead of uploading a new copy
- add tests

## Related issue/s
- https://github.com/ITISFoundation/osparc-simcore-clients/issues/64

### CODE REVIEW

File: clients/python/client/osparc/_utils.py [hunk 1]
```diff
@@ -90,3 +95,15 @@ async def _fcn_to_coro(callback: Callable[..., S], *args) -> S:
     """Get a coroutine from a callback."""
     result = await asyncio.get_event_loop().run_in_executor(None, callback, *args)
     return result
+
+
+def _sha256(file: Path) -> str:
```
Comment (pcrespov) [inline clients/python/client/osparc/_utils.py:100]: MINOR: the module `_utils` is protected but the function is to be used outside of this module. 
I suggest: `_utils.compute_sha256`

File: clients/python/test/e2e/test_files_api.py [hunk 2]
```diff
@@ -3,47 +3,79 @@
...
-            uploaded_file.id, destination_folder=tmp_path
+            uploaded_file1.id, destination_folder=tmp_path
         )
         assert Path(downloaded_file).parent == tmp_path
         assert _hash_file(Path(downloaded_file)) == _hash_file(tmp_file)
-        files_api.delete_file(uploaded_file.id)
+        files_api.delete_file(uploaded_file1.id)
+
+
+@pytest.mark.skipif(
```
Comment (pcrespov) [inline clients/python/test/e2e/test_files_api.py:45]: very nice!

File: clients/python/test/e2e/test_files_api.py [hunk 3]
```diff
@@ -3,47 +3,79 @@
...
+def test_search_files(
+    tmp_file: Path, cfg: osparc.Configuration, use_checksum: bool, use_id: bool
+) -> None:
+    checksum: str = _sha256(tmp_file)
+    assert checksum == _hash_file(tmp_file), "Could not compute correct checksum"
+    results: osparc.PaginationGenerator
+    with osparc.ApiClient(configuration=cfg) as api_client:
+        files_api: osparc.FilesApi = osparc.FilesApi(api_client=api_client)
+        try:
+            results = files_api._search_files(sha256_checksum=checksum)
```
Comment (pcrespov) [inline clients/python/test/e2e/test_files_api.py:60]: Add some line spaces :-) a bit of air :-))))

TIP:
```python
result = do_something()
assert result

result = do_something()
assert result
```

**Review by pcrespov — APPROVED** [review]
cool!
As before, please enrich the title. Here is less critical since the repository as a single purpose thought

File: clients/python/client/osparc/_utils.py (hunk 1 above)
Comment (bisgaard-itis) [reply clients/python/client/osparc/_utils.py:100]: Good point. I have adapted this.

File: clients/python/test/e2e/test_files_api.py (hunk 3 above)
Comment (bisgaard-itis) [reply clients/python/test/e2e/test_files_api.py:60]: done

**Pull request merged**